### DISCUSSION TITLE
Unable to sync Personal Shared Folders

### DISCUSSION THREAD

**Discussion by djjldoooo** [abraunegg/onedrive#2933] [Q&A] [answered]
Hello, 
I'm unable to sync a shared folder. 
The two accounts (sharing and the one I use to see the folders are just regular Microsoft user accounts, not business, if that matters).

I edited both _/home/local_account/.config/onedrive/config_ and _/home/local_account/.config/onedrive/accounts/one_drive_profile/_config and added:
`sync_business_shared_items = "true"`

I also added the shortcut as explained in the documentation. It doesn't have a "world" icon that would match the known issue tracked by: https://github.com/abraunegg/onedrive/issues/966

![image](https://github.com/user-attachments/assets/d6d06a0e-6e12-4cd1-ba02-662ff9283e06)

The command look fine and accepts the _--sync-shared-files_ argument:

```
onedrive --sync --verbose --sync-shared-files  
                                                                           
Reading configuration file: /home/local_account/.config/onedrive/config
Configuration file successfully loaded
Using 'user' configuration path for application config and state data: /home/local_account/.config/onedrive
D-Bus message bus daemon is available; GUI notifications are now enabled
Using IPv4 and IPv6 (if configured) for all network operations
Attempting to contact Microsoft OneDrive Login Service
Successfully reached Microsoft OneDrive Login Service
Checking Application Version ...
Attempting to initialise the OneDrive API ...
Configuring Global Azure AD Endpoints
The OneDrive API was initialised successfully
Opening the item database ...
Application Version:  onedrive v2.5.2
Account Type:         personal
Default Drive ID:     86abbfe1330e65d8
Default Root ID:      86ABBFE1330E65D8!sea8cc6beffdb43d7976fbc7da445c639
Remaining Free Space: 4.99 GB (5362428768 bytes)
Sync Engine Initialised with new Onedrive API instance
All application operations will be performed in the configured local 'sync_dir' directory: /home/local_account/OneDrive
Fetching /delta response from the OneDrive API for Drive ID: 86abbfe1330e65d8
Processing API Response Bundle: 1 - Quantity of 'changes|items' in this bundle to process: 0
Finished processing /delta JSON response from the OneDrive API
No changes or items that can be applied were discovered while processing the data received from Microsoft OneDrive
Performing a database consistency and integrity check on locally stored data
Processing DB entries for this Drive ID: 86abbfe1330e65d8
Processing: ~/OneDrive
The directory has not changed
[...]
The file has not changed
Scanning the local file system '~/OneDrive' for new data to upload
Performing a last examination of the most recent online data within Microsoft OneDrive to complete the reconciliation process
Fetching /delta response from the OneDrive API for Drive ID: 86abbfe1330e65d8
Processing API Response Bundle: 1 - Quantity of 'changes|items' in this bundle to process: 0
Finished processing /delta JSON response from the OneDrive API
No changes or items that can be applied were discovered while processing the data received from Microsoft OneDrive

Sync with Microsoft OneDrive is complete
```

**Comment by abraunegg** [comment, marked as answer]
@djjldoooo 
>```
> Account Type:         personal
> Default Drive ID:     86abbfe1330e65d8
> Default Root ID:      86ABBFE1330E65D8!sea8cc6beffdb43d7976fbc7da445c639
>```

You are using a 'personal' account type .. but you are *trying* to use 'business' account configuration.

When trying to sync 'personal' shared files, all you need to do is to add the shared folder to your account online.

There is nothing else to do - it all should work without extra configuration or setup as the API for Personal Account Types (and that is what you are using) is 100% different as compared to Business Account Types.

Additionally, you have a Microsoft Account Issue - this is denoted by the `sea8cc6beffdb43d7976fbc7da445c639` string in your Root ID. You must raise a support request with Microsoft to resolve this issue.

**Comment by djjldoooo** [comment]
Thanks. I removed my configuration and launched:
```
onedrive --resync --sync --verbose
No user or system config file found, using application defaults
[...]
```

but the folder doesn't appear locally.

By the way I installed and created the profile with the GUI. I don't know if it changes anything.

**Comment by abraunegg** [comment]
@djjldoooo 
> but the folder doesn't appear locally.

You must add the Shared Folder to your OneDrive my using the 'Add Shortcut' link online to add that Shared Folder.

Nothing will sync without you doing that step online.

**Reply by djjldoooo** [reply]
Yes I did that already.

**Comment by abraunegg** [comment]
Your online state under 'My Files' should look like the following:
![image](https://github.com/user-attachments/assets/04c0ecc3-bd92-4935-a6bd-b625972f269c)

If the Shared Folder to sync is not added under 'My Files' it is *not* going to sync.

**Reply by djjldoooo** [reply]
It is:

![image](https://github.com/user-attachments/assets/4413e1ce-6586-43ba-a905-2cb4dc315e01)

![image](https://github.com/user-attachments/assets/c199389f-7e3a-4b1e-a3db-c3fcc7db396e)

Maybe I should remove the GUI profile and do it again. The shared folder shortcut was added later.

**Comment by abraunegg** [comment]
@djjldoooo 
> Maybe I should remove the GUI profile and do it again. The shared folder shortcut was added later.

This has *nothing* to do with a GUI profile.

Perform a manual resync:

```
onedrive -s -v --resync --resync-auth
```

**Reply by djjldoooo** [reply]
The name of the folder doesn't appear::
```
onedrive -s -v --resync --resync-auth

No user or system config file found, using application defaults
Using 'user' configuration path for application config and state data: /home/local_account/.config/onedrive
D-Bus message bus daemon is available; GUI notifications are now enabled
Deleting the saved application sync status ...
Using IPv4 and IPv6 (if configured) for all network operations
Attempting to contact Microsoft OneDrive Login Service
Successfully reached Microsoft OneDrive Login Service
Checking Application Version ...
Attempting to initialise the OneDrive API ...
Configuring Global Azure AD Endpoints
The OneDrive API was initialised successfully
Opening the item database ...
Application Version:  onedrive v2.5.2
Account Type:         personal
Default Drive ID:     86abbfe1330e65d8
Default Root ID:      86ABBFE1330E65D8!sea8cc6beffdb43d7976fbc7da445c639
Remaining Free Space: 4.99 GB (5362428768 bytes)
Sync Engine Initialised with new Onedrive API instance
All application operations will be performed in the configured local 'sync_dir' directory: /home/local_account/OneDrive
Fetching /delta response from the OneDrive API for Drive ID: 86abbfe1330e65d8
Processing API Response Bundle: 1 - Quantity of 'changes|items' in this bundle to process: 8
Finished processing /delta JSON response from the OneDrive API
Processing 7 applicable changes and items received from Microsoft OneDrive
Processing OneDrive JSON item batch [1/1] to ensure consistent local state
Performing a database consistency and integrity check on locally stored data
Processing DB entries for this Drive ID: 86abbfe1330e65d8
Processing: ~/OneDrive
The directory has not changed
Processing: Documents
The directory has not changed
Processing: Documents/xxxxxxxxxxxx.docx
The file has not changed
Processing: Documents/Office Lens
The directory has not changed
Processing: Images
The directory has not changed
Processing: Images/xxxxxxxxxx.odt
The file has not changed
Processing: Pièces jointes
The directory has not changed
Processing: xxxxxxxxxxxx.xlsx
The file has not changed
Scanning the local file system '~/OneDrive' for new data to upload
Performing a last examination of the most recent online data within Microsoft OneDrive to complete the reconciliation process
Fetching /delta response from the OneDrive API for Drive ID: 86abbfe1330e65d8
Processing API Response Bundle: 1 - Quantity of 'changes|items' in this bundle to process: 0
Finished processing /delta JSON response from the OneDrive API
No changes or items that can be applied were discovered while processing the data received from Microsoft OneDrive

Sync with Microsoft OneDrive is complete
Attempting to perform a database vacuum to optimise database
Database vacuum is complete
```

**Comment by abraunegg** [comment]
@djjldoooo 
Looking at the 'logo' for your shared folder it is also not correct ... 

![image](https://github.com/user-attachments/assets/13f7c803-d4f6-4dd8-ba43-7e904d8ce29f)

It needs to look like this:

![image](https://github.com/user-attachments/assets/dd76a975-ebd8-4c38-a135-a2e88a27aecc)


Under the 'Shared' element (your Partage) .. you need to Add the folder correctly to the account you are syncing:

![image](https://github.com/user-attachments/assets/c7bff20b-0733-4545-bc64-8a1bc8412574)

It looks like you have not added this correctly to the account you are using with the Linux OneDrive Client

**Reply by djjldoooo** [reply]
Don't know why I get this icon. I do the same thing as you show in the screenshot. Same icon if I remove the shortcut and do it again :/

**Comment by abraunegg** [comment]
You are probably using the wrong accounts.

Account 1 
* This is the source folder
* Share the Folder with Account 2

Account 2
* Login as Account 2
* View Shared Folder under 'Shared'
* Add to 'My Files' using the Add shortcut to My Files


Ensure the OneDrive Linux Client is configured to authenticate with Account 2 credentials

Follow this process.

It sounds like you are doing this all under Account 1 ...

**Reply by djjldoooo** [reply]
This is what I do. 

I tried to remove the share, do it again, launch the command you gave.
It worked at the beginning:

```
Syncing this OneDrive Personal Shared Folder: Moliets
...
```

I saw the folder being created but then I saw notifications from the GUI saying unable to create the folder, HTTP error... then the folder disappeared. Didn't have time to screenshot the notifications.

If I launch the command again the shared folder is not seen anymore.

**Comment by abraunegg** [comment]
Please stop all systemd services and any GUI service.

Recreate all online shared folders correctly, and then create a debug log file follpwi g this process: https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support

Once you have gathered that, please open a new bug report and email the debug log for analysis.

I am not rulling out a bug, but this sou ds very odd and 100% unable to reproduce.

**Comment by CannonCloud** [comment]
Hi, I have the exact same issue (with the arrow instead of the (-) symbol on the folder), except for I'm just Account 2, and someone has shared a file. Every subfolder from her directory that I "Add shortcut to my files" ends up with the arrow. I don't see a bug report yet. Should I submit it? For reference, I have a free account, and my friend pays for something, maybe Microsoft 365. Thanks!

**Reply by djjldoooo** [reply]
We tried to troubleshoot by email with abraunegg. It's still KO. I think I'm in the same case where account A pays for something and account B has a free account.

Here is its last email:


> Is this is how you are doing it [creating the shortcut as explained in the documentation] – it should be working, but what it is creating is a ‘shortcut’ .. not a ‘link’
> 
> At this point you need to raise a call with Microsoft to seek help – as their website is failing you. I cant help with why this is not working for you in the standard manner that it works for everyone else globally.
> 
> Either there is something weird with your account, your folders , your process … or there is a Microsoft bug.

**Comment by abraunegg** [comment]
> Hi, I have the exact same issue (with the arrow instead of the (-) symbol on the folder), except for I'm just Account 2, and someone has shared a file. Every subfolder from her directory that I "Add shortcut to my files" ends up with the arrow. I don't see a bug report yet. Should I submit it? For reference, I have a free account, and my friend pays for something, maybe Microsoft 365. Thanks!

You need to raise a bug report with Microsoft.

I cannot support and do not support the Microsoft OneDrive Website !

**Suggestions:**

1. Delete all shared references from all account
2. Unshare all folders with all accounts

When testing all of this I use *free* OneDrive Personal Accounts to test sharing of folders between accounts.

I know this feature works without issue.

If you have a *paid* Microsoft OneDrive Personal subscription, raise a case for support with Microsoft.

**Reply by CannonCloud** [reply]
Thanks for the response! If I do figure something out, I'll say what it was.

**Reply by CannonCloud** [reply]
I haven't figured out why the resync doesn't see the shared folders, but I figured out the icon issue. If you switch view to "Tiles" then you get the link icon.

**Comment by abraunegg** [comment]
@CannonCloud 
> I haven't figured out why the resync doesn't see the shared folders, but I figured out the icon issue. If you switch view to "Tiles" then you get the link icon.

Please raise a bug report and generate an application debug log file following this process: https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support

When you run the command, add: `--resync --resync-auth`

The resulting command would be:
```
onedrive --synchronize --verbose --verbose --resync --resync-auth > debug_output.log 2>&1
```

**Comment by abraunegg** [comment]
@djjldoooo 

Please can you read the details in https://github.com/abraunegg/onedrive/issues/2957

Whilst you have other folders / data in your OneDrive Account, this might also be the fix for you - please review the PR portion of my response to potentially resolve your issue.

**Reply by djjldoooo** [reply]
Thanks. I uninstalled the version I had, compiled the one you provided (I'm running Manjaro), ran a --sync, and a --resync but it's still KO :/

**Reply by abraunegg** [reply]
@djjldoooo 
Using the PR, please open a new bug report and provide a new debug log using the following command:
```
onedrive --synchronize --verbose --verbose --resync --resync-auth > debug_output.log 2>&1
```

**Comment by abraunegg** [comment]
@djjldoooo 
Please raise a new bug as requested, with the debug logs provided from PR #2959

**Reply by djjldoooo** [reply]
Hello sorry I wasn't available. Do you still need some debug logs?
I didn't raise a bug at Microsoft as I'm not sure I would be able to explain :)

**Reply by abraunegg** [reply]
The logs would be helpful to compare with  https://github.com/abraunegg/onedrive/issues/2957

I 100% suspect this is a Microsoft issue that you need to raise a fault / bug with Microsoft.

**Comment by cengique** [comment]
@abraunegg I am using one business account and trying to make links from shared Teams and other Sharepoint folders into my account, but they always show up as the arrow icon. Even when I click from the "Shared" section and make a link, it still shows up as a .url file. 
I am using the browser interface of OneDrive. Is there anything else I can try? 
I am on Manjaro with OneDrive version 2.5.3.

**Reply by abraunegg** [reply]
Certain objects will only create .url files. That is a limitation of the Microsoft OneDrive platform.

**Comment by mkasanm** [comment]
This is an ongoing bug which is also affecting many other users:

[OneDrive Shared Folder Syncing Has Been Broken for Months, With No Fix in Sight](https://winbuzzer.com/2025/03/02/onedrive-shared-folder-syncing-has-been-broken-for-months-with-no-fix-in-sight-xcxwbn/)

**Reply by abraunegg** [reply]
@mkasanm 
Interesting read.

The issue is about Microsoft moving users internally from an older platform to a newer platform. This has resulted in 2 bugs:
* https://github.com/OneDrive/onedrive-api-docs/issues/1890
* https://github.com/OneDrive/onedrive-api-docs/issues/1891

These are the cause of all the OneDrive Personal Shared Folder pain.

**Comment by abraunegg** [comment]
@djjldoooo , @mkasanm , @CannonCloud , @cengique 

The Microsoft API Bug has been worked around. Please rebuild your client from 'master' , or wait for the v2.5.5 release.

**Reply by ptrgslr** [reply]
works now for me, thank you so much!
